src/common/internal/test_group.ts

```diff
@@ -411,7 +412,9 @@ class RunCaseSpecific implements RunCase {
         await this.fn(inst as Fixture & { params: {} });
       } finally {
         // Runs as long as constructor succeeded, even if initialization or the test failed.
-        await inst.doFinalize();
+        // Runs async without waiting for results (yet) so that subsequent subcases can start.
+        // All finalization steps will be waited on at the end of the testcase.
+        finalizePromise = inst.doFinalize();
```

Comment:
I see now we're only pipelining the finalize steps, not the actual test bodies. Shouldn't it be possible pipeline the init and test functions too? By basically reverting these changes to runTest() and storing off the whole runTest() promise instead of just the finalizePromise. Guessing there's something that makes that harder than it seems?

Reply:
We could certainly do that - do we want to?

if subcases have `await` inside the test body, then we'll have interleaved subcase execution. Maybe this might be surprising or unexpected for test authors.

Reply:
It should be fine unless they share any state, which tests aren't supposed to (but that's hard to guarantee). I don't know how much of a performance gain we'd get, but there are some tests that do readbacks in the middle that should go faster.

We can start with just the finalizers if we know there's a concrete benefit from it.

Reply:
Well, the implementation is simpler if we go fully async, so I'll do that for now then